core/src/main/java/gov/nist/secauto/metaschema/core/metapath/item/node/INodeItem.java

```diff
@@ -195,6 +210,48 @@ default Stream<? extends INodeItem> descendantOrSelf() {
     return ObjectUtils.notNull(Stream.concat(Stream.of(this), descendant()));
   }
 
+  /**
+   * Get the children of this node's parent that occur after this node in a
+   * depth-first order.
+   *
+   * @return a stream of nodes
+   */
+  @NonNull
+  default Stream<? extends IModelNodeItem<?, ?>> followingSibling() {
+    return ObjectUtils.notNull(Stream.empty());
+  }
+
+  /**
+   * Get the children of this node's parent, and their descendants, that occur
+   * before this node in a depth-first order.
+   *
+   * @return a stream of nodes
+   */
+  @NonNull
+  default Stream<? extends IModelNodeItem<?, ?>> precedingSibling() {
+    return ObjectUtils.notNull(Stream.empty());
+  }
+
+  /**
+   * Get the children of this node's parent, and their descendants, that occur
+   * before this node in a depth-first order.
+   *
+   * @return a stream of nodes
+   */
+  default Stream<? extends IModelNodeItem<?, ?>> following() {
+    return ObjectUtils.notNull(Stream.empty());
+  }
```

Comment:
Non-blocking: I am having trouble understanding how the functions for these different axes all return empty streams, am I misunderstanding that node items do not have preceding and follow siblings?

Reply:
Most node types (i.e. flag, document) do not require returning anything when these methods are called. Onlu IModelNodeItem requires generating these nodes for the axes according to the [Metapath spec](https://www.w3.org/TR/xpath-31/#axes).